x-pack/plugins/actions/common/types.ts

```diff
@@ -34,3 +34,15 @@ export interface ActionTypeExecutorResult<Data> {
   data?: Data;
   retry?: null | boolean | Date;
 }
+
+export function isActionTypeExecutorResult(
+  result: unknown
+): result is ActionTypeExecutorResult<unknown> {
+  const unsafeResult = result as ActionTypeExecutorResult<unknown>;
+  return (
+    unsafeResult &&
+    typeof unsafeResult === 'object' &&
+    unsafeResult?.actionId === 'string' &&
+    (unsafeResult?.status === 'ok' || unsafeResult?.status === 'error')
```

Comment:
This seems fragile, in terms of if we ever change the `status` field to have values other than `ok` | `error`.  I think doing a `typeof ... === 'string'` is probably fine here as well.

Oh, and just noticed the line above is probably intended to be the following (looks like it's missing `typeof`)

```ts
typeof unsafeResult?.actionId === 'string' &&
```

Reply:
Yeah, it's definitely fragile - but I don't think there's another way around this as we do this in a `catch` and Typescript has no type checking on error handling. One of my main gripes with error throwing.

Do you have any advice for another way to do this?

Reply:
It seems like the lines

```ts
unsafeResult?.actionId === 'string' &&
(unsafeResult?.status === 'ok' || unsafeResult?.status === 'error')
```

should be

```ts
typeof unsafeResult?.actionId === 'string' &&
typeof unsafeResult?.status === 'string'
```